### ISSUE TITLE
Maximum Concurrency Queue Event Source For Lambda

### ISSUE THREAD

**Issue by darrencapner** [serverless/serverless#11952]
### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Use case description

Is there a setting to use within the queue or lambda function definition within Serverless.yml file to set the maximum concurrency setting for the queue event source for a lambda trigger that was recently announced by AWS in March of 2023?

[Introducing maximum concurrency of AWS Lambda functions when using Amazon SQS as an event source](https://aws.amazon.com/blogs/compute/introducing-maximum-concurrency-of-aws-lambda-functions-when-using-amazon-sqs-as-an-event-source/)

I can’t find any reference on serverless documentation [Serverless Framework - AWS Lambda Guide - Serverless.yml Reference](https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml)

This configuration is different than reservedConcurrency or provisionedConcurrency on a Lambda.

Added this question to the [Serverless Forum](https://forum.serverless.com/t/maximum-concurrency-queue-event-source-for-lambda/18833)

Thank you

### Proposed solution (optional)

```
max-concurrency-lambda:
   handler: max-concurrency/index.handler
   events:
      - sqs:
          arn: [queue-arn]
          maximumConcurrency: 2
```

**Comment by gustavjonsson1** [comment]
Hey, I think what you're asking for is already available in version 3.27.0 and above.
See https://www.serverless.com/framework/docs/providers/aws/events/sqs

**Comment by darrencapner** [comment]
That it is. Thanks.

My company is still using version 2. So I'll have to move to version 3 to make use of this new feature.
